plugins/flowdock/app/views/samson_flowdock/_notify_buddy_box.html.erb

```diff
@@ -1,29 +1,61 @@
 <% if @deploy.waiting_for_buddy? && @deploy.stage.send_flowdock_notifications? %>
-  <buddy-request-box default-buddy-request-message="<%= default_flowdock_message(@deploy) %>"
-                     flowdock-flows="<%= @deploy.stage.enabled_flows_names.join(', ') %>"
-                     deploy="<%= @deploy.id %>" ></buddy-request-box>
-
-  <script type="text/ng-template" id="templates/buddy_request_box">
-    <div ng-show="shouldDisplayFeedback()">
-      <div class="alert" ng-class="successful ? 'alert-success' : 'alert-danger'">
-        <span>{{ message }}</span>
-      </div>
-    </div>
+  <%= form_tag flowdock_notify_path(deploy_id: @deploy.id+1000) do %>
     <div class="panel panel-info">
       <div class="panel-heading">
-        <h3 class="panel-title">{{ title }}</h3>
+        <h3 class="panel-title">Request a buddy!</h3>
       </div>
       <div class="panel-body">
         <div class="col-sm-10">
           <div class="mentions-input-box">
-            <textarea id="buddy_request_box"></textarea>
+            <%= text_area_tag :message, default_flowdock_message(@deploy), id: :buddy_request_box %>
           </div>
-          <em>This will send a buddy request to the <strong>{{ flowdockFlows }}</strong> flows.</em>
+          <em>This will send a buddy request to the <strong><%= @deploy.stage.enabled_flows_names.join(', ') %></strong> flows.</em>
         </div>
         <div class="col-sm-2">
-          <button class="btn btn-primary" data-ng-click="notifyFlowDock()">Request a Buddy!</button>
+          <%= submit_tag 'Request a Buddy!', class: "btn btn-primary" %>
         </div>
       </div>
     </div>
+  <% end %>
+
+  <script>
+    $(function(){
+      var $input = $('#buddy_request_box');
+
+      // when user hits 'Request', normalize and submit via ajax
+      var $form = $input.parents('form');
+      $form.submit(function(e){
+        e.preventDefault();
+
+        // normalize mentioned users @[alex](user:1) -> @alex
+        $input.mentionsInput('val', function(text){
+          $input.val(text.replace(/@\[([^\\]+?)\]\(([^)]+?)\)/g, "@$1"));
+        });
+
+        // submit it without leaving the page
+        $.ajax({
+          url: $form.attr('action'),
+          type: $form.attr('method'),
+          data: $form.serialize(),
+          success: function() { alert('Message sent!') },
+          error: function(_, _, text) { alert('ERROR:' + text) }
+        });
+      });
+
+      // when user types @, show possible user names
+      var users = <%= SamsonFlowdock::FlowdockService.new(@deploy).users.to_json.html_safe %>;
+      $input.mentionsInput({
+        // prefill with what is already there
+        defaultValue: $input.val(),
+
+        // when user types after the @ filer down the possible users
```

Comment:
or is it ....
[image: Inline image 1]
yeah it is ...

On Tue, Jun 14, 2016 at 1:30 AM, Zehua Liu <notifications@github.com> wrote:

> In plugins/flowdock/app/views/samson_flowdock/_notify_buddy_box.html.erb
> <https://github.com/zendesk/samson/pull/1064#discussion_r66928206>:
>
> > +        $.ajax({
> > +          url: $form.attr('action'),
> > +          type: $form.attr('method'),
> > +          data: $form.serialize(),
> > +          success: function() { alert('Message sent!') },
> > +          error: function(_, _, text) { alert('ERROR:' + text) }
> > +        });
> > +      });
> > +
> > +      // when user types @, show possible user names
> > +      var users = <%= SamsonFlowdock::FlowdockService.new(@deploy).users.to_json.html_safe %>;
> > +      $input.mentionsInput({
> > +        // prefill with what is already there
> > +        defaultValue: $input.val(),
> > +
> > +        // when user types after the @ filer down the possible users
>
> filter typo?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1064/files/bafc898b979eb34b71dcc115de75abb0e7992b51#r66928206>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAsZx_IGxsfey19SXR9-BxAiw9ceOPBks5qLmaNgaJpZM4Iz2PM>
> .
>